lib/model/query/forms.js

```diff
@@ -117,6 +117,42 @@ createNew.audit = (form, partial, _, publish) => (log) =>
     : null));
 createNew.audit.withResult = true;
 
+////////////////////////////////////////////////////////////////////////////////
+// CREATING NEW VERSIONS
+
+// Inserts a new form def into the database for createVersion() below, setting
+// fields on the def according to whether the def will be the current def or the
+// draft def.
+const _createNewDef = (partial, form, publish, data) => async ({ one, enketo, env }) => {
+  const insertWith = (moreData) => one(insert(partial.def.with({
+    formId: form.id,
+    xlsBlobId: partial.xls.xlsBlobId,
+    xml: partial.xml,
+    ...data,
+    ...moreData
+  })));
+
+  if (publish === true) return insertWith({ publishedAt: new Date() });
+
+  // Check whether there is an existing draft that we have access to. If not,
+  // generate a draft token and enketoId.
+  if (form.def.id == null || form.def.id !== form.draftDefId) {
+    const draftToken = generateToken();
+
+    // Try to push the draft to Enketo. If doing so fails or is too slow, then
+    // the worker will try again later.
+    const encodedId = encodeURIComponent(form.xmlFormId);
+    const path = `/v1/test/${draftToken}/projects/${form.projectId}/forms/${encodedId}/draft`;
+    const request = enketo.create(`${env.domain}${path}`, form.xmlFormId);
+    const enketoId = await timebound(request, 0.5).catch(() => null);
```

Comment:
What if `Forms.createVersion()` fails for some reason? Then we will have pushed a form draft to Enketo that we will never use. Is that an issue?

Reply:
I asked about this in Slack. But I don't think it's a big issue.